### ISSUE TITLE
groupPathsBy tag not displaying as nested categories

### ISSUE THREAD

**Issue by pburrows** [PaloAltoNetworks/docusaurus-openapi-docs#1005]
## Describe the bug

Generated API documents are not grouped by tag when rendered. 

## Expected behavior

Documents should be grouped by tag

## Current behavior

*.tag.mdx files are being rendered flat, on the same level with the corresponding API endpoints:

![image](https://github.com/user-attachments/assets/7b8a19af-0f33-47f0-87cc-efa18eb8c7d3)

In the above screenshot, BenefitGroups, BenefitPolicies, and Benefits are all tags with corresponding *.tag.mdx files

## Steps to reproduce

See gist [here](https://gist.github.com/pburrows/b147cc6b1a0d8892a6b488d6b3639b95).

Gist includes the docusaurus config file. The generates sidebar.ts file. The swagger.json file, and an example of one of the generated .tag.mdx files, and one of the generated endpoint mdx files.

An example of the rendered HTML is below. There are no nested `<ul>` tags for any of these `<li>` tags. I expect to see nested `<ul>` tags.

![image](https://github.com/user-attachments/assets/a06ad382-7cc1-4e66-8748-3d8b6d5e30b4)

## Your Environment

versions in package.json:

  "dependencies": {
    "@docusaurus/core": "3.5.2",
    "@docusaurus/preset-classic": "3.5.2",
    "@mdx-js/react": "^3.0.0",
    "bootstrap": "^5.3.3",
    "clsx": "^2.0.0",
    "docusaurus-plugin-openapi-docs": "^4.1.0",
    "docusaurus-plugin-sass": "^0.2.5",
    "docusaurus-theme-openapi-docs": "^4.1.0",
    "prism-react-renderer": "^2.3.0",
    "react": "^18.0.0",
    "react-bootstrap": "^2.10.5",
    "react-dom": "^18.0.0",
    "react-icons": "^5.3.0"
  },

**Comment by sserrata** [comment]
Ah, I must've missed it in the gist - I see it now. Let me take a look...

**Comment by sserrata** [comment]
Ok, I was not able to reproduce the issue.

<img width="335" alt="Screenshot 2024-11-05 at 8 15 53 AM" src="https://github.com/user-attachments/assets/80a5974f-a15e-4349-97ce-030796c6ecbb">

I don't believe you shared it but I would doublecheck your `sidebars.ts` is configured correctly. It's behaving as if you have it set to `autogenerated` instead of pointing to the generated `sidebar.ts` that the plugin provides.

**Comment by pburrows** [comment]
Hm. You are right. I do have it set to autogenerated. Actually, I hadn't changed the default sidebar and it was still called "tutorialSidebar". 

How can I make the generated REST documentation nested inside of this autogenerated sidebar? 
I went and created a second sidebar like in the demo, but when you navigate to that, all the other documentation goes away (its a whole separate sidebar). What I really want is an autogenerated sidebar based on folder structure, but for it to ALSO use the generated sidebar.ts that is in the REST docs folder for that specific folder.

Is there a way to configure that? The docs aren't clear.

**Comment by pburrows** [comment]
I hacked together a solution where I use a category.json file with a className of `hidden` (that correspondingly has CSS with display: none), but it's a pretty bad solution because the HTML is still being sent down the wire and can be seen when you inspect it. 

I also tried generating the API mdx files into a folder that wasn't in the /docs/ path, but that doesn't seem to work when `groupPathsBy: "tag"` is set. Grouping by Tag looks like it requires the files to be in the /docs/ folder for some reason.
